### PR TITLE
Disallow selecting the text of buttons

### PR DESCRIPTION
**Pull request by delvh**
Currently, it is possible to select the text of buttons.
I don't see a reason why, though. A button should (by default) not allow selecting its text.

This PR changes this by adding a new style class that disallows text selection.
In the future, if you want to make something unselectable, simply give it the style class `unselectable`.
If you want every element of another CSS selector to be unselectable, simply use `&:extend(.unselectable);` inside the selector body.

Originated in #19007 but is thematically irrelevant.

This could even be backported as it is a non-breaking change that enhances UX.

### CODE REVIEW

**Comment by silverwind** [comment]
Why not just

```css
button {
  user-select: none;
}
```

Reasons:

1. Keep it simple. If something is not a `<button>`, it will have a11y problems and should be changed to such.
2. Vendor prefixes are not needed, all browsers we support do support the unprefixed variant since many years now.

File: web_src/less/_base.less [hunk 1]
```diff
@@ -1526,12 +1538,12 @@ a.ui.label:hover {
 }
 
 .lines-num {
+  &:extend(.unselectable);
```
Comment (silverwind) [inline web_src/less/_base.less:1538]: `user-select: none;` will be more readable to someone unitiated to Less. Vendor prefixes are not needed.

**Comment by delvh** [comment]
CanIUse gives me only 75% unprefixed.
prefixed, this goes up to 94%.
(https://caniuse.com/?search=user-select)
This would mean that it would not work correctly for **one out of 4** people.
I think that's a bit too much.

Also, there are a lot more pseudo buttons than you think there are.
According to my quick search (`ag 'class=".*button.*".*>' | grep -v '<button' | wc -l`), there are 327 of those... Have fun fixing all of them.

**Comment by silverwind** [comment]
Right, Safari still needs `-webkit-`, so that must be kept, sorry I missed it on MDN.

**Comment by silverwind** [comment]
>  there are 327 of those

Maybe compromise on `button, .button` 😉

**Comment by delvh** [comment]
Yeah, I can definitely throw out the unneeded prefix rules for `-ms` and `-moz`.

That presents me with a problem: There might be a lot more components that might want not to be selectable.
So, which of the following options would you recommend?
- Remove the style class completely, adding its rules to each needing selector (pretty messy to maintain)
- Extend selectors that should not be selectable with the style class (as it is now)
- Remove the `&:extend(.unselectable)` and instead add the style class to every component that qualifies for it
- Did I forgot something?

My personal choice is the current approach because this syntax is not too obscure, and if I'm writing less and see a syntax I don't understand, I can easily google it. To give you an example:
> will be more readable to someone unitiated to Less

That was my first time writing (not CSS compliant) less, I simply googled whether I can extend a styleclass in less and used the first StackOverflow result. So, it should be possible to understand even for newbies.

**Review by 6543 — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Generally L-G-T-M. I prefer to keep only `user-select` and `-webkit-user-select`. Other prefixes are so rare .....

**Comment by wxiaoguang** [comment]
Generally L-G-T-M. I prefer to keep only `user-select` and `-webkit-user-select`. Other prefixes are so rare .....

**Comment by delvh** [comment]
@silverwind, I would like still an answer to the question above.

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
No more comment, will get this PR merged.

**Pull request merged**